app/Console/Services/Importer.php

```diff
@@ -423,4 +460,55 @@ protected function recalculateRanks()
     {
         /*$this->comment('--- RECALCULATING RANKS ---');*/
     }
+
+    /**
+     * Generate user's API Key and email them their new password
+     */
+    protected function setupUsers()
+    {
+        $allusers = User::all();
+        foreach($allusers as $user)
+        {
+            # Generate New User Password
+            # TODO: Increase Security?
+            $newpw = substr(md5(date('mdYhs')), 0, 10);
+
+            # Generate API Key
+            $apikey = Utils::generateApiKey();
+
+            # Update Info in DB
+            $user->password = bcrypt($newpw);
+            $user->api_key = $apikey;
+            $user->save();
+
+            # Send E-Mail to User with new login details
+            $email = new \App\Mail\NewLoginDetails($user, $newpw, 'New Login Details | '.config('app.name'));
+            Mail::to($user->email)->send($email);
+        }
+
+        $this->comment('----All Users E-Mailed!----');
+    }
+
+    /**
+     * Get the user's new state from their original state
+     */
+    protected function getUserState($state)
+    {
+        // Decide which state they will be in accordance with v7
```

Comment:
This looks good... one small change, I like to be explicit. Maybe add an array here,

```php
$phpvms_classic_states = [
  'ACTIVE' => 0,
  ...
];
```

And then check against those.